CefSharp.Core/Internals/ClientAdapter.cpp

```diff
@@ -732,6 +733,16 @@ namespace CefSharp
         bool ClientAdapter::GetAuthCredentials(CefRefPtr<CefBrowser> browser, CefRefPtr<CefFrame> frame, bool isProxy,
             const CefString& host, int port, const CefString& realm, const CefString& scheme, CefRefPtr<CefAuthCallback> callback)
         {
+            if (isProxy && Cef::proxyOptions != nullptr && Cef::proxyOptions->IP == StringUtils::ToClr(host) && !String::IsNullOrEmpty(Cef::proxyOptions->Username)
```

Comment:
> What if you wanted to provide a different username and password after you've initialized CEF?

That's now possible by doing `CefSharpSettings.Proxy.Username = "x"` and `CefSharpSettings.Proxy.Username = "y"`

I don't quite agree with you, I'm hoping this can stay `as is` so that anyone using the `IRequestHandler` doesn't have to implement this logic themself